test/utils/padder/padder.go

```diff
@@ -0,0 +1,202 @@
+func (p *Padder) UntilAvailableReach(resName corev1.ResourceName, quantity string) *Padder {
+	quan := resource.MustParse(quantity)
+	p.untilAvailableReach[resName] = quan
+	return p
+}
+
+// UntilAvailableReachWithResourceList is like UntilAvailableReach but with a complete ResourceList as a parameter
+func (p *Padder) UntilAvailableReachWithResourceList(resources corev1.ResourceList) *Padder {
+	p.untilAvailableReach = resources
+	return p
+}
+
+// Pad will create pad pods in order to align the nodes
+// with the requested amount of available untilAvailableReach
+func (p *Padder) Pad() error {
+	if p.nNodes == 0 {
+		klog.Warningf("no nodes for padding were found. please specify at least one node")
+		return nil
+	}
+
+	nrtList, err := nrtutil.GetUpdated(p.Client, nrtv1alpha1.NodeResourceTopologyList{}, time.Second*10)
+	if err != nil {
+		return err
+	}
+
+	// create a list of only nodes that have single-numa-node policy
+	singleNumaNrt := nrtutil.FilterTopologyManagerPolicy(nrtList.Items, nrtv1alpha1.SingleNUMANodeContainerLevel)
+	singleNumaNrt = append(singleNumaNrt, nrtutil.FilterTopologyManagerPolicy(nrtList.Items, nrtv1alpha1.SingleNUMANodePodLevel)...)
+
+	for i := 0; i < len(singleNumaNrt) && p.nNodes > 0; i++ {
+		nrt := singleNumaNrt[i]
+		nodePadded := false
+		for _, zone := range nrt.Zones {
+			// check that zone has at least the amount of untilAvailableReach that needed
+			if nrtutil.ZoneResourcesMatchesRequest(zone.Resources, p.untilAvailableReach) {
+				availResList := nrtutil.AvailableFromZone(zone)
+				diffList, err := diffAvailableToExpected(availResList, p.untilAvailableReach)
+				if err != nil {
+					return err
+				}
+
+				// create a pod that asks for exact amount of untilAvailableReach as the diff
+				// in order to reach to desired amount of available untilAvailableReach in the node
+				padPod := objects.NewTestPodPause(p.Namespace, fixture.RandomName("padder"))
+
+				// label the pod with a pad label, so it will be easier to identify later
+				labelPod(padPod, map[string]string{PadderLabel: ""})
+
+				padPod.Spec.Containers[0].Resources.Limits = diffList
+				padPod.Spec.Containers[0].Resources.Requests = diffList
+
+				// place the pod on the selected node
+				//padPod.Spec.NodeSelector = map[string]string{nodes.LabelHostname: nrt.Name}
+				padPod.Spec.NodeSelector = map[string]string{"kubernetes.io/hostname": nrt.Name}
+				if err := p.Client.Create(context.TODO(), padPod); err != nil {
+					return err
+				}
+				klog.InfoS("created pod", "pod", fmt.Sprintf("%s/%s", padPod.Namespace, padPod.Name), "node", nrt.Name)
+				nodePadded = true
+			} else {
+				klog.Warningf("node: %q zone: %q, doesn't have enough available untilAvailableReach", nrt.Name, zone.Name)
+			}
+		}
+		if nodePadded {
+			p.nNodes--
+		}
+	}
+	return nil
+}
+
+// Clean can be called after test finished
+// in order to clean all the padding pod in an easier way
```

Comment:
if we create the padding pod in yet another namespace managed by the `Padder`, we can just delete the parent namespace (wait for the pod to be gone so the resources are released) and call it a day

Reply:
The namespace is not managed by the padder, it was given to it by the user.

Think of a case when there are other components inside the namespace.
We don't want to delete them behind the back of the user.
We only want to delete the pods we care about- i.e. the padding pod.

Reply:
correct, but what's the use case for letting the client manage the namespace?
padding pods are logically part of the test fixture. What's the benefit in NOT having the padder package manage the namespace?
If needed (can't see the usecase yet, though) the padder package can return the relevant namespace and the label(s) needed to get the padding pods in the padding namespace

Reply:
I still tend to see the padding pods as a fixture implementation detail, hence I see little merit but some drawbacks in expecting a namespace instead of self-managing, but is not wrong either. Let's start this way and let's see how it works out.